src/app.js

```diff
@@ -27,18 +28,24 @@ function onDidReceiveSettings(json) {
   sendQuery(json.context)
 }
 
+function onWillDisappear(json) {
+  console.info('onWillDisappear', json)
+  const context = json.context
+  clearInterval(contextState[context].intervalID)
+}
+
 function onWillAppear(json) {
   console.info('onWillAppear', json)
   const context = json.context
   setSettings(context, json.payload.settings)
   contextState[context].canvas = new Canvas(context)
   sendQuery(context)
-  setInterval(() => {
-    sendQuery(context)
-  }, FETCH_INTERVAL * 1000)
+  contextState[context].intervalID = setInterval(sendQuery, FETCH_INTERVAL * 1000, context)
+
 }
 
 function onKeyDown(json) {
+  console.log('onKeyDown', json);

```

Comment:
Logging on every keystroke seems is too verbose IMO

Reply:
Removed